test/sampling.jl

```diff
@@ -5,6 +5,26 @@ import StatsBase: norepeat, randi
 
 srand(1234)
 
+# test that rng specification is working correctly
+# a) if the same rng is passed to a sample function twice, 
+#    the results should be the same (repeatability)
+# b) not specifying a rng should be the same as specifying Base.GLOBAL_RNG
+function test_rng_use(func, non_rng_args...)
+    # some sampling methods mutate a passed array and return it
+    # so that the tests don't pass trivially, we need to copy those
+    # pre-allocated storage arrays
+
+    # repeatability
+    @test func(MersenneTwister(1), deepcopy(non_rng_args)...) ==
```

Comment:
Sorry, I hadn't noticed this was called on the tuple of arguments rather than directly on the array. That's OK then.